src/jit/simdcodegenxarch.cpp

```diff
@@ -619,8 +619,8 @@ CodeGen::genSIMDIntrinsicInit(GenTreeSIMD* simdNode)
             // data section using movss/sd.  Similarly if op1 is a memory op we
             // would have loaded it using movss/sd.  Movss/sd when loading a xmm reg
             // from memory would zero-out upper bits. In these cases we can
-            // avoid explicitly zero'ing out targetReg.
-            bool zeroInitRequired = !(op1->IsCnsFltOrDbl() || op1->isMemoryOp());
+            // avoid explicitly zero'ing out targetReg if targetReg and op1Reg are the same.
+            bool zeroInitRequired = (targetReg != op1Reg) || !(op1->IsCnsFltOrDbl() || op1->isMemoryOp());
```

Comment:
On further thinking...

I am not sure that i understand why zero init is required if targetReg != op1Reg?

As per my understanding if op1 is a float constant or a memory-op, we would have loaded using movss/movsd, in which case upper bits of op1Reg would be zero.  Then genSIMDScalarMove() would have moved entire 128/256-bits of op1reg to targetReg using movaps instruction, which would mean even upper bits of targetReg would be zero initialized.

What am i missing?